### DISCUSSION TITLE
Contributor's Guide Improvements

### DISCUSSION THREAD

**Discussion by grigorescu** [zeek/zeek#1229] [Documentation]
I wonder if people can share a bit about their development environment for Zeek core development. Editors, tools that they use, tips and tricks to make it easier for people to get started or avoid common pitfalls.

The only Zeek-specific optimizations that I've really done are a couple of helper scripts, to install Zeek into a branch-specific prefix, and a script to quickly toggle between different versions.

I saw some commits earlier referencing VSCode. I have yet to try it, but keep hearing good things.

The [contributor's guide](https://docs.zeek.org/en/current/devel/contributors/index.html) right now is kind of sparse, so I'm hoping that based on some of the tips and tricks collected here, that can be fleshed out a bit more.

**Comment by rsmmr** [comment]
We should also link to https://github.com/zeek/zeek/wiki/Contribution-Guide, and vice versa, or maybe even combine.

**Reply by grigorescu** [reply]
Ohh, right, thanks. I knew I had seen that, but couldn't find it. I think linking or combining would help.

I think in that guide, we should also highlight a couple of things we'd like to see for every PR, e.g. `make test && clang-tidy`. They're in the guide, and captured by the automated PR checks, but making them more obvious means they're more likely to be done.

**Comment by timwoj** [comment]
I use emacs for editing and command-line tools for compiling and debugging (lldb, mostly). Occasionally I will use CLion if I need the visual debugger bits. I've tried switching to other tools in the past but I usually find myself coming back to emacs.

I can share my .emacs if someone wants it. There was some talk about writing a major-mode for zeek scripts at one point, but I don't think anyone ever actually took it on.

**Comment by jsiwek** [comment]
The main tools I rely on: `vim`, `vim-lsp`, `clangd`.  Also `gdb`/`lldb`, but I've more recently found VS Code to work well for debugging tasks.  VS Code also has Good Vim emulation from what I've found so far -- personally reluctant to spend time fully configuring and completely switching over for general text-editing, but seems to be a viable option for any Vim-inclined.
